taskmanager/taskq.go

```diff
@@ -89,76 +96,88 @@ func (c *Client) loadTaskQ() error {
 	return nil
 }
 
-// registerTaskUsingTaskQ will register a new task using the TaskQ engine
-func (c *Client) registerTaskUsingTaskQ(task *Task) {
-
+// RegisterTask will register a new task using the TaskQ engine
+func (c *TaskManager) RegisterTask(name string, handler interface{}) (err error) {
 	defer func() {
-		if err := recover(); err != nil {
-			c.DebugLog(fmt.Sprintf("registering task panic: %v", err))
+		if panicErr := recover(); panicErr != nil {
+			err = fmt.Errorf(fmt.Sprintf("registering task panic: %v", panicErr))
 		}
 	}()
 
 	mutex.Lock()
+	defer mutex.Unlock()
 
-	// Check if task is already registered
-	if t := taskq.Tasks.Get(task.Name); t != nil {
-
-		// Register the task locally
-		c.options.taskq.tasks[task.Name] = t
+	if t := taskq.Tasks.Get(name); t != nil {
+		// if already registered - register the task locally
```

Comment:
maybe instead of comment, we could have log message here, like:
> task {taskname} registered in taskq engine, by different instance

Reply:
In standard application flow - the task registration is at the initialization stage so the re-register condition never happens. I left this option (which existed before by changes) because it broke many tests.

Reply:
I thought that if we use Redis, then taskq.Tasks.Get(name) can return info that there is already such task registered. but maybe my assumptions was wrong

Reply:
I'll double check it. Because now, I'm not quite sure :)

Reply:
I checked it. Every instance have to do it's own registration and for every instance the `taskq.Tasks.Get(name)` returns nil at the initialization stage. So I think - an additional log is not needed.

![image](https://github.com/BuxOrg/bux/assets/152964795/fdb473cd-0d30-4124-876f-efeffcc21318)